tests/unit/test_generate_umm_var.py

```diff
@@ -0,0 +1,276 @@
+                                for record in generated_umm_var])
+
+        self.assertSetEqual(actual_variables, expected_variables)
+
+        # Check that no attempt was made to publish a UMM-Var record to CMR:
+        mock_publish_umm_var.assert_not_called()
+
+    @patch('varinfo.umm_var.publish_umm_var')
+    @patch('varinfo.cmr_search.GranuleQuery')
+    @patch('varinfo.generate_umm_var.download_granule')
+    def test_generate_collection_umm_var_with_publication(self,
+                                                          mock_download_granule,
+                                                          mock_granule_query,
+                                                          mock_publish_umm_var):
+        """ A request with all the necessary information should succeed. """
+        expected_concept_ids = [
+            'V0000000001-PROV', 'V0000000002-PROV', 'V0000000003-PROV',
+            'V0000000004-PROV', 'V0000000005-PROV', 'V0000000006-PROV',
+            'V0000000007-PROV', 'V0000000008-PROV'
+        ]
+        mock_granule_query.return_value.get.return_value = [
+            {
+                'links': [
+                    {
+                        'href': self.netcdf4_url,
+                        'rel': 'http://esipfed.org/ns/fedsearch/1.1/data#'
+                    }
+                ]
+            }
+        ]
+
+        # Add side effect that will copy test file to the temporary directory,
+        # simulating a download.
+        mock_download_granule.side_effect = self.download_granule_side_effect
+
+        mock_publish_umm_var.side_effect = expected_concept_ids
+
+        # Run the test:
+        published_umm_var = generate_collection_umm_var(
+            self.collection_concept_id,
+            self.bearer_token_header,
+            publish=True
+        )
+
+        # Ensure the granule query used expected query parameters
+        mock_granule_query.return_value.parameters.assert_called_once_with(
+            downloadable=True,
+            sort_key='-start_date',
+            concept_id=self.collection_concept_id
+        )
+
+        # Ensure the call to download the granule had correct parameters
+        mock_download_granule.assert_called_once_with(self.netcdf4_url,
+                                                      self.bearer_token_header,
+                                                      out_directory=ANY)
+
+        # Ensure the output looks as expected
+        self.assertSetEqual(set(published_umm_var),
+                            set(expected_concept_ids))
+
+    @patch('varinfo.umm_var.publish_umm_var')
+    @patch('varinfo.cmr_search.GranuleQuery')
+    @patch('varinfo.generate_umm_var.download_granule')
+    def test_error_from_search_is_raised(self, mock_download_granule,
+                                         mock_granule_query,
+                                         mock_publish_umm_var):
+        """ Ensure an error raised iduring search is propagated out to the
```

Comment:
Typo with `during` and some other stuff... I think you wanted to say:
`""" Ensure an error is raised during search and it is propagated out to the ...`

Reply:
- [ ] Fix typo